src/platform/process.cpp

```diff
@@ -0,0 +1,130 @@
+#include "process.h"
+#include <stdlib.h>
+#include <algorithm>
+#include <boost/predef.h>
+#include <boost/filesystem.hpp>
+#include <boost/assert.hpp>
+#if( BOOST_OS_WINDOWS )
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#else
+#include <unistd.h>
+#endif
+
+namespace platform {
+namespace process {
+#if( BOOST_OS_WINDOWS )
+namespace buffers {
+size_t const CommandLineMax = 1024;
+wchar_t CommandLine[ CommandLineMax ] = { 0 };
+}
+namespace {
+std::wstring wcstr( std::string const& str )
+{
+    size_t len = mbstowcs( NULL, str.c_str(), 0 );
+    std::wstring wstr( len + 1, '\0' );
+    mbstowcs( &wstr[0], str.c_str(), wstr.size() );
+    return wstr;
+}
+}
+bool Start( std::string const& command, std::vector<std::string> const& arguments )
+{
+    std::vector<std::wstring> args;
+    std::transform( arguments.begin(), arguments.end(), std::back_inserter( args ), &wcstr );
+    return Start( wcstr( command ), args );
+}
+bool Start( std::wstring const& command, std::vector<std::wstring> const& arguments )
+{
+    wchar_t* pt = buffers::CommandLine;
+    size_t s = command.size();
+    memcpy( pt, command.c_str(), s * sizeof( wchar_t ) );
+    pt += s - 1;
+    *( pt++ ) = L' ';
+    for( auto const& arg : arguments )
```

Comment:
Sure could, however
accumulate( args.begin(), args.end(), pt, [](wchar_t* pt, std::string const& arg ) { loop_core; return pt; } );
seems less readable to me. Although I think I'll drop the whole copy-to-preallocated-buffer idea, accumulate is definitely better for simple wstring to wstring concat. (The reasoning behind that was to minimize allocations after a crash, but first of all screw that, secondly the string->wstring conversion already allocates quite some data. And getting rid of that would be pretty damn ugly.)